sdk/storage/Azure.Storage.Blobs/src/Generated/Models/FilterBlobItem.cs

```diff
@@ -26,19 +27,13 @@ internal FilterBlobItem(string name, string containerName)
             {
                 throw new ArgumentNullException(nameof(containerName));
             }
+            if (tags == null)
+            {
+                throw new ArgumentNullException(nameof(tags));
+            }
```

Comment:
what's the call hierarchy of this ctor? Is this param required by public surface?

Reply:
This is an internal constructor on an internal class.

Reply:
Yes I can read that. Which public API calls this?

Reply:
`BlobServiceClient.FilterBlobsByTag()`.  But the only call to that constructor is `FilterBlobItem.DeserializeFilterBlobItem(XElement element)`.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/bdb426a844ff84ff822d9d1fe8e97f8bb9da8279/sdk/storage/Azure.Storage.Blobs/src/Generated/Models/FilterBlobItem.Serialization.cs#L17-L32
This looks capable of producing null if service won't send tags back.

Reply:
@tasherif-msft did you make `Tags` required in `FilterBlobItem`?

Reply:
Previously we had 
![image](https://user-images.githubusercontent.com/69483382/124652415-f9862480-de50-11eb-9ac3-0134ee5969e5.png)

"TagValue" seemed to have no effect and hence changed to `Tags` which is actually defined in the swagger. It does appear when those blobs are filtered they would have the Tags key.
Are we supposed to have neither @seanmcc-msft?

Reply:
@tasherif-msft, could you remove `Tags` from the required list?

I believe what happened is that the service changed the tag format.  https://docs.microsoft.com/en-us/rest/api/storageservices/find-blobs-by-tags#response-body

Before, it looked like this:
```
<EnumerationResults ServiceEndpoint="https://seanmcccanary3.blob.core.windows.net/">
	<Where>&quot;myTagKey&quot;=&apos;myTagValue&apos;</Where>
	<Blobs>
		<Blob>
			<Name>test-blob-7ed5124d-b15b-48bf-af4f-fcdeb1b910d3</Name>
			<ContainerName>test-container-e6eb39d9-606b-4718-866c-6345d31ede7f</ContainerName>
			<TagValue>myTagValue</TagValue>
		</Blob>
	</Blobs>
	<NextMarker/>
</EnumerationResults>
```
Now, it looks like this:
```
<EnumerationResults ServiceEndpoint="https://seanmcccanary3.blob.core.windows.net/">
	<Where>&quot;myTagKey&quot;=&apos;myTagValue&apos;</Where>
	<Blobs>
		<Blob>
			<Name>test-blob-42b6ab05-00eb-4e36-99cd-de051c66bf69</Name>
			<ContainerName>test-container-08a80321-8e9e-4164-bf38-3d43f2a183d5</ContainerName>
			<Tags>
				<TagSet>
					<Tag>
						<Key>myTagKey</Key>
						<Value>myTagValue</Value>
					</Tag>
				</TagSet>
			</Tags>
		</Blob>
	</Blobs>
	<NextMarker/>
</EnumerationResults>

```

Reply:
Yeah, we should make it not required. If somebody sends older service version then they won't even get `<Tags>` in the xml doc.

Reply:
@seanmcc-msft alright so I will be removing Tags then

Reply:
Fixed.